### PR TITLE
Update PHPUnit and remove unnecessary code left behind after updating PHP minimum required version

### PR DESCRIPTION
**Pull request by ste93cry**
As per title, I've updated PHPUnit to the greatest version still supported on PHP 7.2 and I cleaned up some leftovers of the update of the minimum required version of PHP

### CODE REVIEW

File: tests/Util/JSONTest.php [hunk 1]
```diff
@@ -100,49 +99,6 @@ public function encodeSubstitutesInvalidUtf8CharactersOnPhp72OrGreaterDataProvid
...
-     * @requires PHP < 7.2
-     *
-     * @dataProvider encodeSubstitutesInvalidUtf8CharactersOnPhp71OrLowerDataProvider
-     */
-    public function testEncodeSubstitutesInvalidUtf8CharactersOnPhp71OrLower($value, string $expectedResult): void
-    {
-        $this->assertSame($expectedResult, JSON::encode($value));
-    }
-
-    public function encodeSubstitutesInvalidUtf8CharactersOnPhp71OrLowerDataProvider(): \Generator
```
Comment (Jean85) [inline tests/Util/JSONTest.php:113]: Dropping this should mean that we could improve the `Json` class by using native escaping, right?

File: composer.json [hunk 2]
```diff
@@ -47,8 +47,8 @@
         "phpstan/extension-installer": "^1.0",
         "phpstan/phpstan": "^0.12",
         "phpstan/phpstan-phpunit": "^0.12",
-        "phpunit/phpunit": "^7.5.18",
-        "symfony/phpunit-bridge": "^4.3|^5.0",
+        "phpunit/phpunit": "^8.5",
```
Comment (Jean85) [inline composer.json:50]: Should we allow PHPUnit 9? So we could test under PHP 8 later

**Review by Jean85 — CHANGES_REQUESTED** [review]

File: tests/Util/JSONTest.php (hunk 1 above)
Comment (ste93cry) [reply tests/Util/JSONTest.php:113]: You're right, I cleaned up the tests but not the code :pensive:

**Review by Jean85 — APPROVED** [review]

**Review by stayallive — APPROVED** [review]
Looks good 👍

**Pull request closed without merging**